PrivateCloud.DiagnosticInfo/PrivateCloud.DiagnosticInfo.psm1

```diff
@@ -710,6 +712,35 @@ function Start-CopyJob(
     }
 }
 
+# 
+# Utility wrapper for invoking commands by opening sessions  
+# for each of the cluster nodes and preserving the session  
+# to be deleted after use. 
+# 
+
+function Invoke-SddcCommonCommand ( 
+    [ValidateNotNullOrEmpty()][string[]] $ClusterNodes = @(), 
+    [string] $JobName,
+	[scriptblock] $InitBlock, 
+    [scriptblock] $ScriptBlock 
+    ) 
+{ 
+	$Job        = @() 
+	$Sessions   = @() 
+	$SessionIds = @() 
+
+	$Sessions = New-PSSession -ComputerName $ClusterNodes 
+	Invoke-Command -Session $Sessions $InitBlock
+	$Job      = Invoke-Command -Session $Sessions -AsJob -JobName $JobName -ScriptBlock $ScriptBlock 
+
+	foreach ($s in $Sessions) { 
+		$SessionIds += $s.Id 

```

Comment:
You can replace this foreach with a simple "$SessionIds = $Sessions.Id". Slicing a list with a member specifier is a very natural (and simplifying) powershell idiom. Its used many other places in the module.